core/server/data/migrations/versions/4.14/02-fix-free-members-status-events.js

```diff
@@ -1,36 +1,59 @@
+const _ = require('lodash');
 const logging = require('@tryghost/logging');
 const {createTransactionalMigration} = require('../../utils.js');
 
 module.exports = createTransactionalMigration(
     async function up(knex) {
         logging.info('Updating members_status_events for free members');
-        const freeMembers = await knex('members')
-            .select('id')
-            .where('status', 'free');
+        const freeMemberEvents = await knex('members')
+            .select(
+                'members.id as member_id',
+                'members_status_events.id',
+                'members_status_events.to_status',
+                'members_status_events.created_at'
+            )
+            .innerJoin(
+                'members_status_events',
+                'members.id',
+                'members_status_events.member_id'
+            )
+            .where('members.status', 'free');
 
-        if (freeMembers.length === 0) {
+        if (freeMemberEvents.length === 0) {
             logging.info('No free members found - skipping migration');
             return;
-        } else {
-            logging.info(`Found ${freeMembers.length} free members - checking members_status_events`);
         }
 
-        for (const member of freeMembers) {
-            const mostRecentStatusEvent = await knex('members_status_events')
-                .select('*')
-                .where('member_id', member.id)
-                .orderBy('created_at', 'desc')
-                .limit(1)
-                .first();
-
-            if (!mostRecentStatusEvent) {
-                logging.warn(`Could not find a status event for member ${member.id} - skipping this member`);
-            } else if (mostRecentStatusEvent.to_status !== 'free') {
-                logging.info(`Updating members_status_event ${mostRecentStatusEvent.id}`);
-                await knex('members_status_events')
-                    .update('to_status', 'free')
-                    .where('id', mostRecentStatusEvent.id);
+        const eventsByMember = _.groupBy(freeMemberEvents, 'member_id');
+
+        const eventsToUpdate = Object.keys(eventsByMember).reduce((incorrectEvents, memberId) => {
+            const events = eventsByMember[memberId];
+
+            events.sort((a, b) => {
+                return new Date(b) - new Date(a);
```

Comment:
Should this be `b.created_at` and `a.created_at`?

Reply:
EUGH YES!